src/databricks/sql/auth/retry.py

```diff
@@ -0,0 +1,378 @@
+
+        urllib3 calls Retry.new() between successive requests as part of its `.increment()` method as shown below:
+
+        ```python
+            new_retry = self.new(
+                total=total,
+                connect=connect,
+                read=read,
+                redirect=redirect,
+                status=status_count,
+                other=other,
+                history=history,
+        )
+        ```
+
+        The arguments it passes to `.new()` (total, connect, read, etc.) are those which modified by `.increment()`.
+
+        Since our subclass has a new __init__ signature and requires custom state variables, we override the method
+        to pipe our Databricks-specific state while preserving the super-class's behaviour.
+        """
+
+        # These arguments will match the function signature for self.__init__
+        databricks_init_params = dict(
+            delay_min=self.delay_min,
+            delay_max=self.delay_max,
+            stop_after_attempts_count=self.stop_after_attempts_count,
+            stop_after_attempts_duration=self.stop_after_attempts_duration,
+            delay_default=self.delay_default,
+            force_dangerous_codes=self.force_dangerous_codes,
+            _retry_start_time=self._retry_start_time,
+            _command_type=self._command_type,
+            urllib3_kwargs={},
+        )
+
+        # Gather urllib3's current retry state _before_ increment was called
+        # These arguments match the function signature for super().__init__
+        # Note: if we update urllib3 we may need to add/remove arguments from this dict
+        urllib3_init_params = dict(
+            total=self.total,
+            connect=self.connect,
+            read=self.read,
+            redirect=self.redirect,
+            status=self.status,
+            other=self.other,
+            allowed_methods=self.allowed_methods,
+            status_forcelist=self.status_forcelist,
+            backoff_factor=self.backoff_factor,  # type: ignore
+            backoff_max=self.backoff_max,  # type: ignore
+            raise_on_redirect=self.raise_on_redirect,
+            raise_on_status=self.raise_on_status,
+            history=self.history,
+            remove_headers_on_redirect=self.remove_headers_on_redirect,
+            respect_retry_after_header=self.respect_retry_after_header,
+            backoff_jitter=self.backoff_jitter,  # type: ignore
+        )
+
+        # Update urllib3's current state to reflect the incremented counters
+        urllib3_init_params.update(**urllib3_incremented_counters)
+
+        # Include urllib3's current state in our __init__ params
+        databricks_init_params["urllib3_kwargs"].update(**urllib3_init_params)  # type: ignore
+
+        return type(self)(
+            **databricks_init_params,  # type: ignore[arg-type]
+        )
+
+    @property
+    def command_type(self) -> Optional[CommandType]:
+        return self._command_type or None
```

Comment:
Why is this necessary?  It seems like this only evaluates the right hand of the or if its already None.

Reply:
Good catch. This was a holdover from during development.

Fixed a38a2859789cf73f513aa1af70c6bf5aa676e0a6